src/main/java/teammates/ui/controller/InstructorFeedbackEditCopyAction.java

```diff
@@ -0,0 +1,140 @@
+package teammates.ui.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.common.util.StringHelper;
+import teammates.logic.api.GateKeeper;
+
+public class InstructorFeedbackEditCopyAction extends Action {
+
+    @Override
+    protected ActionResult execute() throws EntityDoesNotExistException {        
+        String newFeedbackSessionName = getRequestParamValue(Const.ParamsNames.COPIED_FEEDBACK_SESSION_NAME);
+        String[] coursesIdToCopyTo = getRequestParamValues(Const.ParamsNames.COPIED_COURSES_ID);
+        String originalFeedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        String originalCourseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+        
+        Assumption.assertNotNull("null course id", originalCourseId);
+        Assumption.assertNotNull("null fs name", originalFeedbackSessionName);
+        Assumption.assertNotNull("null copied fs name", newFeedbackSessionName);
+
+        
+        if (coursesIdToCopyTo == null || coursesIdToCopyTo.length == 0) {
+            return createRedirectToEditPageWithErrorMsg(originalFeedbackSessionName, originalCourseId, Const.StatusMessages.FEEDBACK_SESSION_COPY_NONESELECTED);
+        }
+        
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(originalCourseId, account.googleId); 
+        
+        new GateKeeper().verifyAccessible(
+                instructor, 
+                logic.getCourse(originalCourseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION);
```

Comment:
Is there a check that the instructor has permissions for the courses to copy to?